gems/pending/util/esx_thumb_print.rb

```diff
@@ -0,0 +1,34 @@
+require 'util/thumb_print'
+
+class ESXThumbPrint < ThumbPrint
+  attr_reader :user, :password, :cert
+
+  def initialize(host, user, password)
+    @user     = user
+    @password = password
+    super(host)
+  end
+
+  def uri
+    url  = "https://" + @host + "/host/ssl_cert"
+    @uri = URI(url)
+    @uri
+  end
+
+  def http_request
+    super
+    @request          = Net::HTTP::Get.new(@uri.request_uri)
```

Comment:
Calling `@uri` here may end up being a nil if uri itself is never called.  You can ensure it's called by using uri as an accessor and just calling `uri.request_uri` here.  Then you don't need the instance variables either.